frontend/userpatch.lua

```diff
@@ -0,0 +1,154 @@
+--[[--
+Allows applying developer patches and shell scripts while running KOReader.
+
+The contents in `koreader/userpatches/` are applied on calling `userpatch.applyPatches(priority)`.
+--]]--
+
+local isAndroid, android = pcall(require, "android")
+
+local userpatch =
+    {   -- priorities for user patches,
+        early_once = "0",  -- to be started early on startup (once after an update)
+        early = "1",       -- to be started early on startup (always, but after an `early_once`)
+        late = "2",        -- to be started after UIManager is ready (always)
+                           -- 3-7 are reserved for later use
+        before_exit = "8", -- to be started a bit before exit before settings are saved (always)
+        on_exit = "9",     -- to be started right before exit (always)
+
+        applyPatches = function(priority) end, -- to be overwritten, if the device allows it.
+    }
+
+if isAndroid and android.prop.flavor == "fdroid" then
+    return userpatch
+end
+
+------------------------------------------------------------------------------------
+
+local lfs = require("libs/libkoreader-lfs")
+local logger = require("logger")
+
+-- We can not use util.lua so early during startup.
+local function removeTrailingSlash(str)
+    if str:sub(-1, -1) == "/" then
+        return str:sub(1, -2)
+    end
+    return str
+end
+-- the directory KOReader is installed in (and runs from)
+local package_dir = removeTrailingSlash(lfs.currentdir():match("^.*/"))
+-- the directory where KOReader stores user data (on SDL we may set `XDG_DOCUMENTS_DIR=~/koreader/`)
```

Comment:
the directory where KO stores app data (writable and available to users) is [`data_dir`](https://github.com/koreader/koreader/blob/master/datastorage.lua#L10-L35)

You're using its parent here as home dir but limited to some platforms. Please use `DataStorage` and, if needed, get the root path from `data_dir` instead.

Also, please don't name it `home_dir` where user puts document, not application data.

Reply:
Good point, thanks.